internal/ioctl/ioctl.go

```diff
@@ -161,6 +174,85 @@ func GetCBTInfo() ([]types.CBTInfo, error) {
 	return cbtInfoRet, nil
 }
 
+func getKernelEntryAddress(name string) (uint64, error) {
+	var address uint64
+	execPath, err := os.Executable()
+	if err != nil {
+		return address, errors.Wrap(err, "getting the path of the executable")
+	}
+	out, err := exec.Command("sudo", execPath, "-ksym-address", name).Output()
```

Comment:
An alternative to using sudo to read the actual addresses is to set the ```CAP_SYSLOG``` capability on the binary:

```bash
sudo setcap 'CAP_SYSLOG+ep' /usr/local/bin/coriolis-snapshot-agent
```

You only need to do this once, then the binary will be able to read those kernel entries without sudo. You can leave this in if you want, either way is fine.